consensus/src/main/java/org/ethereum/beacon/consensus/transition/BlockTransition.java

```diff
@@ -0,0 +1,167 @@
+          slashable_vote_data.custody_bit_0_indices + slashable_vote_data.custody_bit_1_indices.
+       Let intersection = [x for x in indices(slashable_vote_data_1) if x in indices(slashable_vote_data_2)].
+       For each validator index i in intersection run penalize_validator(state, i)
+           if state.validator_registry[i].penalized_slot > state.slot.
+    */
+    for (CasperSlashing casper_slashing : block.getBody().getCasperSlashings()) {
+      Set<UInt24> indices_1 = new HashSet<>();
+      indices_1.addAll(asList(casper_slashing.getSlashableVoteData1().getCustodyBit0Indices()));
+      indices_1.addAll(asList(casper_slashing.getSlashableVoteData1().getCustodyBit1Indices()));
+      Set<UInt24> indices_2 = new HashSet<>();
+      indices_2.addAll(asList(casper_slashing.getSlashableVoteData2().getCustodyBit0Indices()));
+      indices_2.addAll(asList(casper_slashing.getSlashableVoteData2().getCustodyBit1Indices()));
+      Set<UInt24> intersection = indices_1;
+      intersection.retainAll(indices_2);
+      for (UInt24 index : intersection) {
+        if (state
+                .getValidatorRegistry()
+                .get(index.getValue())
+                .getPenalizedSlot()
+                .compareTo(state.getSlot())
+            > 0) {
+          specHelpers.penalize_validator(state, index.getValue());
+        }
+      }
+    }
+
+    /*
+       Attestations
+
+       For each attestation in block.body.attestations:
+       Append PendingAttestationRecord(
+           data=attestation.data,
+           participation_bitfield=attestation.participation_bitfield,
+           custody_bitfield=attestation.custody_bitfield,
+           slot_included=state.slot
+       ) to state.latest_attestations.
+    */
+    for (Attestation attestation : block.getBody().getAttestations()) {
+      PendingAttestationRecord record =
+          new PendingAttestationRecord(
+              attestation.getData(),
+              attestation.getParticipationBitfield(),
+              attestation.getCustodyBitfield(),
+              state.getSlot());
+      state.withNewLatestAttestation(record);
+    }
+
+    /*
+       Deposits
+
+       For each deposit in block.body.deposits:
+       Run the following:
+
+       process_deposit(
+           state=state,
+           pubkey=deposit.deposit_data.deposit_input.pubkey,
+           amount=deposit.deposit_data.amount,
+           proof_of_possession=deposit.deposit_data.deposit_input.proof_of_possession,
+           withdrawal_credentials=deposit.deposit_data.deposit_input.withdrawal_credentials,
+           randao_commitment=deposit.deposit_data.deposit_input.randao_commitment,
+           custody_commitment=deposit.deposit_data.deposit_input.custody_commitment,
+       )
+    */
+    final ValidatorRegistryUpdater registryUpdater =
+        ValidatorRegistryUpdater.fromState(state, spec);
+    for (Deposit deposit : block.getBody().getDeposits()) {
+      registryUpdater.processDeposit(deposit);
```

Comment:
Do we want to maintain both `SpecHelpers` and `RegistryUpdater`? Maybe abandon `ValidatorRegistryUpdater` for a while and handle everything in `SpecHelpers`?

Reply:
Yep, makes sense for now